cpp/src/gandiva/CMakeLists.txt

```diff
@@ -146,6 +150,7 @@ if (ARROW_GANDIVA_BUILD_TESTS)
   add_gandiva_unit_test(lru_cache_test.cc)
   add_gandiva_unit_test(to_date_holder_test.cc to_date_holder.cc date_utils.cc)
   add_gandiva_unit_test(simple_arena_test.cc)
+  add_gandiva_unit_test(decimal_type_util_test.cc decimal_type_util.cc)
```

Comment:
I'm curious why not just link the test files with libgandiva.a? This is similar to the `ADD_PARQUET_TEST` macro we have here https://github.com/apache/arrow/blob/master/cpp/src/parquet/CMakeLists.txt#L28

Reply:
Sure, no problem